src/core/init/get_discontinuities.ts

```diff
@@ -31,56 +25,49 @@ const { BUFFER_DISCONTINUITY_THRESHOLD } = config;
 
 /**
  * Perform various checks about discontinuities during playback.
- * @param {Observable} clock$
+ * @param {Object} tick
  * @param {Object} manifest
- * @returns {Observable}
+ * @returns {Array | undefined}
  */
 export default function getDiscontinuities(
-  clock$: Observable<IInitClockTick>,
+  tick: IInitClockTick,
   manifest: Manifest
-) : Observable< [ number, number ] > {
-  return clock$.pipe(
-    filter(({ stalled }) => stalled !== null),
-    map((tick) : [number, number] | undefined => {
-      const { buffered, currentTime, currentRange, state, stalled } = tick;
-      const nextBufferRangeGap = getNextRangeGap(buffered, currentTime);
-      // 1: Is it a browser bug? -> force seek at the same current time
-      if (isPlaybackStuck(currentTime,
-                          currentRange,
-                          state,
-                          stalled !== null)
-      ) {
-        log.warn("Init: After freeze seek", currentTime, currentRange);
-        return [currentTime, currentTime];
+) : [ number, number ] | undefined {
+  const { buffered, currentTime, currentRange, state, stalled } = tick;
+  const nextBufferRangeGap = getNextRangeGap(buffered, currentTime);
+  // 1: Is it a browser bug? -> force seek at the same current time
+  if (isPlaybackStuck(currentTime,
+                      currentRange,
+                      state,
+                      stalled !== null)
+  ) {
+    log.warn("Init: After freeze seek", currentTime, currentRange);
+    return [currentTime, currentTime];
```

Comment:
Isn't that included in the bug we want to fix here? I'm not sure but I would guess so